src/ng/http.js

```diff
@@ -1088,7 +1103,11 @@ function $HttpProvider() {
         }
 
         // send request
-        return sendReq(config, reqData).then(transformResponse, transformResponse);
+        try {
+          return sendReq(config, reqData).then(transformResponse, transformResponse);
+        } catch (e) {
+          return $q.reject(e);
```

Comment:
Actually this is needed. We are not trying to catch exceptions from the `then` handlers, we are catching exceptions thrown synchronously inside `sendReq`. E.g. `url = $sce.getTrustedResourceUrl(url);` can throw.